tests/core/actions/test_forms.py

```diff
@@ -1334,3 +1359,218 @@ async def test_ask_for_slot_if_not_utter_ask(
+            ["chitchat", "greet"],
+            ["chitchat"],
+            "from_text",
+        ),
+        # same examples for entity_type -> from_intent
+        (
+            # `ignored_intents` as a string and slot's not_intent as an empty list.
+            "greet",
+            [],
+            "from_intent",
+        ),
+        (
+            # `ignored_intents` as an empty list and slot's not_intent has a value.
+            [],
+            ["greet"],
+            "from_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value different than the ones in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["inform"],
+            "from_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value that is included also in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["chitchat"],
+            "from_intent",
+        ),
+        # same examples for entity_type -> from_trigger_intent
+        (
+            # `ignored_intents` as a string and slot's not_intent as an empty list.
+            "greet",
+            [],
+            "from_trigger_intent",
+        ),
+        (
+            # `ignored_intents` as an empty list and slot's not_intent has a value.
+            [],
+            ["greet"],
+            "from_trigger_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value different than the ones in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["inform"],
+            "from_trigger_intent",
+        ),
+        (
+            # `ignored_intents` as a list of 2 values and slot's not_intent has one
+            # value that is included also in `ignored_intents`.
+            ["chitchat", "greet"],
+            ["chitchat"],
+            "from_trigger_intent",
+        ),
+    ],
+)
+def test_ignored_intents(
+    ignored_intents: Union[Text, List[Text]],
+    slot_not_intent: Union[Text, List[Text]],
+    type: Text,
+):
+    form_name = "some_form"
+    entity_name = "some_slot"
+    form = FormAction(form_name, None)
+
+    if type == "from_entity":
+        domain = Domain.from_dict(
+            {
+                "forms": {
+                    form_name: {
+                        IGNORED_INTENTS: ignored_intents,
+                        REQUIRED_SLOTS_KEY: {
+                            entity_name: [
+                                {
+                                    "type": type,
+                                    "entity": entity_name,
+                                    "not_intent": slot_not_intent,
+                                }
+                            ],
+                        },
+                    }
+                }
+            }
+        )
+    elif type == "from_text":
```

Comment:
why don't we make the form input paramatrizable (the dictionary which is currently mapped to `form_name`? The `if type == ...` make the test hard to read and to understand. An alternative could also be to split the test

Reply:
That is so True! Very good point Tobias!!!! 👌🏾 I am going to split it.